packages/tests/test/reya_common/trade/PassivePool.fork.c.sol

```diff
@@ -255,4 +266,288 @@ contract PassivePoolForkCheck is BaseReyaForkTest {
+                            receiverAddress: sec.periphery
+                        })
+                    );
+
+                    vm.warp(block.timestamp + 1);
+                }
+            }
+        }
+    }
+
+    function check_autoRebalance_currentTargets() public {
+        autoRebalancePool();
+    }
+
+    function check_autoRebalance_differentTargets() public {
+        vm.prank(sec.multisig);
+        IPassivePoolProxy(sec.pool).setAllocationConfiguration(
+            sec.passivePoolId, AllocationConfigurationData({ quoteTokenTargetRatio: 0.353535e18 })
+        );
+
+        address[] memory supportingCollaterals =
+            IPassivePoolProxy(sec.pool).getQuoteSupportingCollaterals(sec.passivePoolId);
+        uint256[] memory newSupportingCollateralsAllocations = new uint256[](supportingCollaterals.length);
+
+        newSupportingCollateralsAllocations[newSupportingCollateralsAllocations.length - 2] = 0.454545e18;
+        newSupportingCollateralsAllocations[newSupportingCollateralsAllocations.length - 1] = 1e18 - 0.454545e18;
+
+        vm.prank(sec.multisig);
+        IPassivePoolProxy(sec.pool).setAllocations(sec.passivePoolId, newSupportingCollateralsAllocations);
+
+        autoRebalancePool();
+    }
+
+    function check_autoRebalance_noSharePriceChange() public {
+        uint256 sharePrice0 = IPassivePoolProxy(sec.pool).getSharePrice(sec.passivePoolId);
+        check_autoRebalance_differentTargets();
+        uint256 sharePrice1 = IPassivePoolProxy(sec.pool).getSharePrice(sec.passivePoolId);
+        assertLe(sharePrice0, sharePrice1);
+        assertApproxEqAbsDecimal(sharePrice0, sharePrice1, 1e8, 18);
+    }
+
+    function check_autoRebalance_maxExposure() public {
+        (uint256 maxExposureShort0, uint256 maxExposureLong0) =
+            IPassivePerpProxy(sec.perp).getPoolMaxExposures(sec.passivePoolId);
+
+        check_autoRebalance_differentTargets();
+
+        (uint256 maxExposureShort1, uint256 maxExposureLong1) =
+            IPassivePerpProxy(sec.perp).getPoolMaxExposures(sec.passivePoolId);
+
+        assertNotEq(maxExposureShort0, maxExposureShort1);
+        assertNotEq(maxExposureLong0, maxExposureLong1);
+        assertApproxEqRelDecimal(maxExposureShort0, maxExposureShort1, 0.075e18, 18);
+        assertApproxEqRelDecimal(maxExposureLong0, maxExposureLong1, 0.075e18, 18);
+    }
+
+    function check_autoRebalance_instantaneousPrice() public {
+        uint128 marketId = 1;
+        uint256 instantaneousPrice0 = IPassivePerpProxy(sec.perp).getInstantaneousPoolPrice(marketId);
+        check_autoRebalance_differentTargets();
+        uint256 instantaneousPrice1 = IPassivePerpProxy(sec.perp).getInstantaneousPoolPrice(marketId);
+        assertNotEq(instantaneousPrice0, instantaneousPrice1);
+        assertApproxEqRelDecimal(instantaneousPrice0, instantaneousPrice1, 0.075e18, 18);
+    }
+
+    function check_sharePriceChangesWhenAssetPriceChanges() public {
+        autoRebalancePool();
```

Comment:
hmm, just thought about another interesting invariant we could check -> make sure that the share price increase in percentage terms is always smaller or equal to the percentage supporting token price, this should ensure that the pool does not "over-report" increases in the value of its assets

hmm, thinking in the other direction, if the asset price were to decline, we'd want the pool share price to decline at least as much in percentage terms